File: Current/Product/Production/Common/Common.Tasks/WaitForExit.vb
Imports System.Threading
Imports System.Collections
Imports System.IO
Imports System.Text
Imports NAnt.Core
Imports NAnt.Core.Tasks
Imports NAnt.Core.Attributes
Imports System



<TaskName("waitforexit")> _
Public Class WaitForExit
    Inherits Task

    Private _taskNames As StringList

    <BuildElement("tasknames", Required:=True)> _
    Public Property TaskNames() As StringList
        Get
            Return _taskNames
        End Get
        Set(ByVal Value As StringList)
            _taskNames = Value
        End Set
    End Property

    Protected Overrides Sub ExecuteTask()
        Dim WorkerThread As Thread
        For Each TaskName As String In Me.TaskNames.StringItems.Values
            WorkerThread = AsyncExecList.Item(TaskName)
            If Not WorkerThread Is Nothing AndAlso WorkerThread.IsAlive Then
                WorkerThread.Join()
            End If
        Next
    End Sub
End Class

File: Product/Production/Nant/CIFactory.NAnt.Tasks/Tasks/WaitForExit.cs
using System;
using System.Text;
using System.IO;
using System.Diagnostics;
using System.Collections;
using System.Collections.Generic;
using NAnt.Core;
using NAnt.Core.Tasks;
using NAnt.Core.Attributes;
using CIFactory.NAnt.Types;


namespace CIFactory.NAnt.Tasks
{
    [TaskName("waitforexit")]
    public class WaitForExit : Task
    {
        #region Fields

        private StringList _taskNames;

        #endregion

        #region Properties

        [BuildElement("tasknames", Required = true)]
        public StringList TaskNames
        {
            get { return _taskNames; }
            set { _taskNames = value; }
        }

        #endregion

        #region Protected Methods

        protected override void ExecuteTask()
        {
            AsyncExec Worker;
            foreach (string TaskName in this.TaskNames.StringItems.Values)
            {
                Worker = AsyncExecList.GetTask(TaskName);
                Worker.Wait();
            }
        }

        #endregion

    }

}